src/core/server/server.ts

```diff
@@ -149,6 +146,9 @@ export class Server {
       context: contextServiceSetup,
     });
 
+    // setup i18n prior to any other service, to have translations ready
+    const i18nServiceSetup = await this.i18n.setup({ http: httpSetup, pluginPaths });
```

Comment:
Had to move this after the `http` service setup to register the route. I could move i18n initialization in another method (like we did for `plugin.discover`) to keep the i18n setup before context and http, but I don't think this is really necessary